lib/jwt/jwk/key_finder.rb

```diff
@@ -37,22 +36,36 @@ def key_for(kid)
       # Returns the key for the given token
       # @param [JWT::EncodedToken] token the token
       def call(token)
-        key_for(token.header['kid'])
+        kid = token.header['kid']
+        x5t = token.header['x5t']
+        x5c = token.header['x5c']
+
+        if kid
+          key_for(kid, :kid)
+        elsif x5t
+          key_for(x5t, :x5t)
+        elsif x5c
```

Comment:
what about the `x5t#S256`header? Should we include it also?

Reply:
I don't have a good idea how `x5t#S256` will be returned in JWK params (the payload should contain a field `x5t#S256` but I wonder how will that be parsed) so I don't add that for now. I could add it quickly if you think it's better to add that.

Reply:
Ok, if it's unclear lets leave it out.